app/services/puppet_class_importer.rb

```diff
@@ -179,7 +179,7 @@ def proxy_environments
   end
 
   def to_be_created_environments
-    proxy_environments - Environment.where(:name => proxy_environments).pluck(:name)
+    proxy_environments - Environment.unscoped.where(:name => proxy_environments).pluck(:name)
```

Comment:
I added the `unscoped` exactly because of the error. When you have no environments and you get the error, import just seems broken to the user. You are also leaking information out of taxonomy scope - now you know there is an environment that you are not supposed to see and you know its name.

Import should not try to create duplicates IMHO, because resources are shared. So if there is already an env in db but we do not have it in our taxonomy, then we are not supposed to see it or touch it.